bin/generate-version-file.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
```

Comment:
hey @gregtatum, how comfortable are you about using a shell script instead of a node script?

My preference still goes to the node script. Advantages are:
* better portability
* can require package.json to access its inner data (the shell script relies on `$npm_*` variables, so needs to be run by `yarn`)
* I still need to do more work around tags later, and I'm a bit afraid about doing the logic in shell code.